### DISCUSSION TITLE
Any wallet available in react native?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#186] [Questions] [answered]
### Describe your question in detail.

Asked in discord dev-discussion

> What wallet is supported in react native having trouble finding one that works for bare react native, want to be able to connect users to their aptos wallet mobile apps

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0x-j** [comment]
I don't think we support connecting to Petra mobile or other mobile wallet in a react native app. But we are working on a keyless solution which lets user login with their google account. It's possible to implement this in your app. 

Here's a [tutorial on integrating keyless](https://learn.aptoslabs.com/example/aptogotchi-keyless/intro), it's still in testnet stage. We'd love to get your feedback.

**Comment by 0x-j** [comment, marked as answer]
Petra has a doc on how to use deeplink to connect to it in your app. Let us know if you have more questions.

https://petra.app/docs/mobile-deeplinks#mobile-2-mobile-communication

**Reply by sujathasperi2022** [reply]
This allows only to communicate through deep links. There isn't any support package or any available to integrate on ReactNative
